pkgs/build-support/setup-hooks/auto-patchelf.py

```diff
@@ -336,7 +336,7 @@ def main() -> None:
     )
     parser.add_argument(
         "--extra-args",
-        nargs="*",
+        nargs="...",
```

Comment:
I could not find any documentation for "..." passed to nargs. What does it do ? How is it different from "*" ?

Reply:
It's a weird "hidden" feature - it basically means "interpret the rest of the arguments as values for this option", so `--extra-args --beep --boop` is parsed as `extra_args = ["--beep", "--boop"]`.

Reply:
That is also the semantics of "*", that is where I am confused.

Oh, I get it, this argument needs to receive arguments itself, as with `--no-clobber-old-sections` above, and argparse chokes on these. 

What about an `action="append"` and passing `--extra-args=--no-clobber-old-sections` instead ? Then  `--extra-args=` can be used multiple times, to escape the strange values. Because using "..." force you to use --extra-args as the last option, otherwise it also swallows valid argparse options. Like `--extra-args --no-clobber-old-sections --ignore-missing` will pass `--ignore-missing` to patchelf instead of passing it to the python script.

Reply:
It's harder to escape the arguments properly in the bash wrapper if doing things that way. Right now it's just a bash array that gets expanded cleanly.